### PR TITLE
First batch of React error fixes

### PR DESCRIPTION
**Pull request by martinpitt**
This starts to address or at least confine the React errors that were spotted by fixing the console.error() checks in PR #1269

 - [x] Land https://github.com/cockpit-project/cockpit/pull/19537 and update to it

### CODE REVIEW

**Comment by martinpitt** [comment]
[This error](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1270-20231026-134109-32e5fa30-fedora-38-devel/log.html#24) is really awkward. `AddDiskModalBody` has a very async way of initializing its `this.state.format`, via `componentDidMount` → `Promise` → `initialState`. So the initial render() will always run with it undefined. But also, `getDefaultVolumeFormat()` can easily return `undefined`.  Fixing this requires someone with knowledge how this should be fixed *properly*. I tried a hack like this:
```diff
--- src/components/vm/disks/diskAdd.jsx
+++ src/components/vm/disks/diskAdd.jsx
@@ -377,6 +377,7 @@ export class AddDiskModalBody extends React.Component {
             customDiskVerificationMessage: null,
             verificationInProgress: false,
             file: null,
+            format: undefined,
         };
         this.onValueChanged = this.onValueChanged.bind(this);
         this.dialogErrorSet = this.dialogErrorSet.bind(this);
@@ -399,6 +400,7 @@ export class AddDiskModalBody extends React.Component {
                     .map(pool => ({ name: pool.name, type: pool.type }))
                     .sort(sortFunction)[0];
 
+        console.log("XXX AddDiskModalBody initialState; defaultPool", defaultPool, "format", defaultPool ? getDefaultVolumeFormat(defaultPool) : "none");
         return {
             storagePools,
             vm,
@@ -412,7 +414,7 @@ export class AddDiskModalBody extends React.Component {
             existingVolumeName: undefined,
             size: 1,
             unit: units.GiB.name,
-            format: defaultPool && getDefaultVolumeFormat(defaultPool),
+            format: defaultPool ? getDefaultVolumeFormat(defaultPool) : "none",
             target: availableTarget,
             permanent: !domainIsRunning(vm.state), // default true for a down VM; for a running domain, the disk is attached tentatively only
             hotplug: domainIsRunning(vm.state), // must be kept false for a down VM; the value is not being changed by user
@@ -732,6 +734,9 @@ export class AddDiskModalBody extends React.Component {
     }
 
     render() {
+        if (this.state.format === undefined) // componentDidMount → initialState did not run yet
+            return;
+
         const Dialogs = this.context;
         const { dialogLoading, vm, storagePools, vms } = this.state;
         const { isMediaInsertion } = this.props;

```

which helps a bit, but breaks the test.

This React warning is really useful and points out an actual bug.

File: src/components/common/needsShutdown.jsx [hunk 1]
```diff
@@ -159,7 +159,10 @@ export const VmNeedsShutdown = ({ vm }) => {
             position="bottom"
             hasAutoWidth
             bodyContent={body}>
-            <Label className="resource-state-text" href="javascript:null" color="blue" id={`vm-${vm.name}-needs-shutdown`} icon={<PendingIcon />}>{_("Changes pending")}</Label>
+            <Label className="resource-state-text" color="blue" id={`vm-${vm.name}-needs-shutdown`}
+                   icon={<PendingIcon />} onClick={() => null}>
```
Comment (cockpituous) [inline src/components/common/needsShutdown.jsx:163]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1270-20231027-113557-7f5e6075-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline src/components/common/needsShutdown.jsx:163]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1270-20231027-115214-e7bf5c8f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (jelly) [inline src/components/common/needsShutdown.jsx:163]: Wondering if dropping `onClick` is enough?

**Review by jelly — APPROVED** [review]
Thanks for the cleanups!

File: src/components/common/needsShutdown.jsx (hunk 1 above)
Comment (martinpitt) [reply src/components/common/needsShutdown.jsx:163]: I *added* it. It needs to have either a `href=` or an `onClick` handler to become focusable, and I think that was the original intent for that hack.

**Pull request merged**